x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectRealm.java

```diff
@@ -412,6 +417,15 @@ public String toString() {
             return name;
         }
 
+        private static void validateClaimType(Object claimValueObject, String settingKey) {
+            if (claimValueObject instanceof String == false &&
```

Comment:
I'd prefer a more descriptive method name if possible. The current feels like it will validate the claim value is a certain specified type. But it really just asserts the value is either:
* A String or
* If it is a collection, it must be a collection of String

Now I am not sure if this is what we really want? Is it OK if the value is a `Number`?

Reply:
Hey can you elaborate on what you mean be "descriptive" here? The method does what it intends to do as described in the PR description:

> When mapping claims to user properties ( principal, email, groups,
name), we only handle string and array of string type. Previously
we would fail to recognize an array of other types and that would
cause failures when trying to cast to String.

Doing so, seems to me to be "Validating the type of the claim value" and as such `validateClaimType` seems appropriate. What would be an alternative that you are more comfortable with ?

Reply:
What I mean is the name gave me the first impression that the signature would be something like:
```java
void validateCliamType(Object value, Class<?> type)
```
But instead the type is not specified but should always be String or collection of it. So I'd personally prefer something like `validateParsableClaim`.

Reply:
Fair enough, I don't necessarily agree that validateClaimType off :) but I don't dislike `validateParsableClaim` enough to fortify my hill :)

Reply:
Thanks Ioannis. The more important piece of my comment here is about the behaviour, i.e. the `if` check does not seem to do what we want to ensure, i.e. `String` or `Collection<String>`:

```java
claimValueObject instanceof String == false && 
claimValueObject instanceof Collection && 
((Collection) claimValueObject).stream().allMatch(c -> c instanceof String) == false
```

If the value is a `Number`, the `if` branch will not be executed, i.e. a `Number` will be considered as parsable.

Reply:
Aha! Now I saw it , thanks, will address